client/sites/hooks/use-transfer-with-software-start-mutation.ts

```diff
@@ -0,0 +1,64 @@
+import { useMutation, UseMutationResult } from '@tanstack/react-query';
+import wpcom from 'calypso/lib/wp';
+
+type transferWithSoftwareResponse = {
+	transferId: number;
+};
+
+type SoftwareSlug = string;
+type SoftwareStatus = 'install' | 'activate';
+type Software = Record< SoftwareSlug, SoftwareStatus >;
+
+type transferOptions = {
+	siteId: number;
+	from?: string;
+	plugins?: Software;
+	themes?: Software;
+};
+
+const requestTransferWithSoftware: (
+	transferOptions: transferOptions
+) => Promise< transferWithSoftwareResponse > = async ( { siteId, from, plugins, themes } ) => {
+	const response = await wpcom.req.post(
+		{
+			path: `/sites/${ siteId }/atomic/transfer-with-software?http_envelope=1`,
+		},
+		{
+			apiNamespace: 'wpcom/v2',
+			body: { plugins, themes, settings: { migration_source_site_domain: from } },
```

Comment:
I think we should also receive the `settings` as an argument to the function instead of hardcoding the `migration_source_site_domain` option here. It should come from the upper levels, especially since this is specific for the migration flow.

Reply:
That makes sense, let me know if this looks good type-wise!